README.md

```diff
@@ -1,38 +1,155 @@
+- [📖 About](#-about)
+- [🛠️ Getting Started](#️-getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+- [🧪 Running Tests](#-running-tests)
+  - [Run Tests](#run-tests)
+  - [Reset Tests](#reset-tests)
+- [🚀 Usage](#-usage)
+  - [Adding SNOS as a Dependency](#adding-snos-as-a-dependency)
+  - [Using the **prove_block** Binary](#using-the-prove_block-binary)
+- [🤝 Related Projects](#-related-projects)
+- [📚 Documentation](#-documentation)
+- [📜 License](#-license)
+
+
+## 📖 About
+
+[Starknet OS](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/core/os/os.cairo) is a [Cairo](https://www.cairo-lang.org/) program designed to prove the integrity of state transitions between blocks on Starknet.
+
+By re-executing transactions from a block and verifying consistency, it produces a [PIE](https://github.com/starkware-libs/cairo-lang/blob/a86e92bfde9c171c0856d7b46580c66e004922f3/src/starkware/cairo/lang/vm/cairo_pie.py#L219-L225) (Program Independent Execution) result. This PIE can be used to generate a Stark proof of integrity, which, if accepted by Starknet L1 verifiers, confirms block validity and updates the Starknet state root in the [StarknetCore contract](https://etherscan.io/address/0xc662c410c0ecf747543f5ba90660f6abebd9c8c4#code).
+
+## 🛠️ Getting Started
 
-## Test Setup
+### Prerequisites
 
-**Cairo Env**
-(see: https://docs.cairo-lang.org/0.12.0/quickstart.html)
+Ensure you have the following dependencies installed:
+- [Rust 1.76.0 or newer](https://www.rust-lang.org/tools/install)
+- Cargo (included with Rust)
+- [pyenv](https://github.com/pyenv/pyenv-installer?tab=readme-ov-file#install) (recommended for managing Python versions)
 
+### Installation
+
+1. **Clone the Repository**
+
+Clone this repository and its submodules:
+   ```bash
+   git clone https://github.com/keep-starknet-strange/snos.git --recursive
+  ``` 
+
+#### Install project dependencies
+In order to compile the Starknet OS Cairo program, you’ll need the Cairo compiler:
+
+- Follow the [Cairo documentation](https://docs.cairo-lang.org/quickstart.html)
+- Or simply run:
 ```bash
-poetry install
-poetry shell
+./setup-scripts/setup-cairo.sh
 ```
 
-**Run Tests**
+This will create a virtual environment and download needed dependencies to compile cairo programs. You will need to activate it to compile Cairo programs.
+
+## 🧪 Running Tests
+To verify your setup, follow these steps:
+
+### Activate snos-env:
+```bash
+ source ./snos-env/bin/activate
+ ```
+
+### Run Tests
 
 ```bash
 ./scripts/setup-tests.sh
 
 cargo test
 ```
 
-**Reset Tests**
+### Reset Tests
+
+If you need to reset the test environment:
 
 ```bash
 ./scripts/reset-tests.sh
 ```
+
+##  🚀 Usage 
+### Adding SNOS as a dependency
+
+You can add the following to your rust project's `Cargo.toml`:
+
+```toml
+starknet-os = { git = "https://github.com/keep-starknet-strange/snos", rev = "662d1706f5855044e52ebf688a18dd80016c8700" }
```

Comment:
we can add that when we add SNOS into crates.io
For now this is more as a placeholder

Reply:
> we can add that when we add SNOS into crates.io

When will that be?

It takes 2 seconds to create a tag at that commit. Otherwise it doesn't look very professional to ask users to use a "random" commit.

Reply:
I would say that it makes sense to add a tag and point to a specific commit when we finish debugging and have 100% success in Sepolia.
This will happen in around 3 weeks and its when we will make our first release